.travis.yml

```diff
@@ -31,14 +33,25 @@ before_install:
 - sudo /etc/init.d/memcached stop || true
 - sudo /etc/init.d/mysql stop || true
 - sudo /etc/init.d/postgresql stop || true
+- cat /etc/nix/nix.conf > ~/nix.conf
+- echo "binary-caches = https://cache.nixos.org https://hydra.iohk.io" >> ~/nix.conf
+- echo "binary-cache-public-keys = hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=" >> ~/nix.conf
+- export NIX_CONF_DIR=~
+
+install:
+  - source scripts/set_nixpath.sh
 
 after_failure:
 - sudo dmesg
 
 script:
 # DEVOPS-107: wait 20min for output due to haddock generation
-- if [[ "$HLINT" == "true" ]]; then
+- if [[ "$MODE" == "nix" ]]; then
+    nix-build -A cardano-sl-static --no-build-output --keep-going || nix-build -A cardano-sl -j 1 ;
```

Comment:
the parallelism enabled by default makes the logs impossible to read (all output is interleaved)
`--no-build-output` solves that, but now you cant see the error from the last job that failed
so in the event of failures, run it again without `--no-build-output -j1`

that should probably be fixed within nix itself, as a new option